cmd/juju/charmhub/info.go

```diff
@@ -103,6 +106,20 @@ func (c *infoCommand) Init(args []string) error {
 	default:
 		return errors.Errorf("unexpected unicode flag value %q, expected <auto|never|always>", c.unicode)
 	}
+
+	switch c.arch {
+	case "all", "amd64", "arm64", "ppc64", "s390":
```

Comment:
We might want to move this into a core package (supported arches or something like that) and use a set.Contians instead of explicitly testing for individual arch names.

Reply:
Also, instead of `"all"` we should use `ArchAll`